File: EDD 2024-2/Parcial_Practico.py/PP1.cpp
#include<vector>
#include<sstream>
#include<iostream>
#include<bits/stdc++.h>

using namespace std;
int main(){
    ios::sync_with_stdio(false); cin.tie(0);
    string comm;
    int prestigio;
    string input;
    vector<int> v={};
    make_heap(v.begin(), v.end());

    while(true){ 
        getline(cin>>ws, input);
        istringstream str(input);
        str>>comm;

        if(comm == "fin") break;
        else if(comm == "salida"){
            if(!v.empty()){
                pop_heap(v.begin(), v.end());
                v.pop_back();
                cout<<"hasta pronto"<<endl;
            }
        }else if(comm == "ingresa"){
            str >> prestigio;
            if(v.empty() || prestigio >= ceil((float)v[0])/2){
                v.push_back(prestigio);
                push_heap(v.begin(), v.end());
                cout<<"adelante"<<endl;
            }else{
                cout<<"denegado"<<endl;
            }
        }
    }
}

File: EDD 2024-2/Parcial_Practico.py/PP1.py
import heapq
club = []
while True:
    N = input().split()
    if N[0] == "fin":
        break

    if N[0] == "ingresa":
        P = -int(N[1])

        if not club:
            heapq.heappush(club, P)
            print("adelante")
 
        elif (club[0]/2) >= (P):
            heapq.heappush(club, P)
            print("adelante")

        else:
            print("denegado")

 
    elif N[0] =="salida":
        if club:
            heapq.heappop(club)
            print("hasta pronto")